test/e2e/opaque_resource.go

```diff
@@ -0,0 +1,309 @@
+			Expect(err).NotTo(HaveOccurred())
+			// TODO(CD): Watch scheduler events instead to catch an indication of
+			//           insufficient resources or other unschedulable condition
+			//           instead of waiting for this to time out.
+			Expect(pod).To(notBeScheduled())
+		})
+
+		It("should account opaque integer resources in pods with multiple containers.", func() {
+			By("Creating a pod with two containers that together require less of the opaque resource than is allocatable on a node")
+			requests := api.ResourceList{opaqueResName: resource.MustParse("1")}
+			limits := api.ResourceList{}
+			// This pod consumes 2 "foo" resources.
+			pod := &api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name: "mult-container-oir",
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "pause",
+							Image: framework.GetPauseImageName(f.Client),
+							Resources: api.ResourceRequirements{
+								Requests: requests,
+								Limits:   limits,
+							},
+						},
+						{
+							Name:  "pause-sidecar",
+							Image: framework.GetPauseImageName(f.Client),
+							Resources: api.ResourceRequirements{
+								Requests: requests,
+								Limits:   limits,
+							},
+						},
+					},
+				},
+			}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(pod)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pod).To(beScheduled())
+
+			By("Creating a pod with two containers that together require more of the opaque resource than is allotable on any node")
+			requests = api.ResourceList{opaqueResName: resource.MustParse("3")}
+			limits = api.ResourceList{}
+			// This pod consumes 6 "foo" resources.
+			pod = &api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Name: "mult-container-over-max-oir",
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "pause",
+							Image: framework.GetPauseImageName(f.Client),
+							Resources: api.ResourceRequirements{
+								Requests: requests,
+								Limits:   limits,
+							},
+						},
+						{
+							Name:  "pause-sidecar",
+							Image: framework.GetPauseImageName(f.Client),
+							Resources: api.ResourceRequirements{
+								Requests: requests,
+								Limits:   limits,
+							},
+						},
+					},
+				},
+			}
+			pod, err = f.Client.Pods(f.Namespace.Name).Create(pod)
+			Expect(err).NotTo(HaveOccurred())
+			// TODO(CD): Watch scheduler events instead to catch an indication of
+			//           insufficient resources or other unschedulable condition
+			//           instead of waiting for this to time out.
+			Expect(pod).To(notBeScheduled())
+		})
+	})
+})
+
+// Returns the current version of the supplied node. If `old` is nil, returns
+// any non-master node.
+func refresh(f *framework.Framework, old *api.Node) *api.Node {
+	if old == nil {
+		// Priming invocation; select the first non-master node.
+		nodes, err := f.Client.Nodes().List(api.ListOptions{})
```

Comment:
I think you can probably refactor to use the informer for all list/watch notify checks.